### PR TITLE
Cleanup release-1.17 slack mergewarnings and milestone applier

### PR DESCRIPTION
**Pull request by cpanato**
### Cleanup
 - remove release branch 1.17 from the slack mergewarnings 
 ~- remove release-1.17 jobs~
 - remove milestone applier for release-1.17

since this branch is not receiving updates anymore.

/kind cleanup

/assign @justaugustus @saschagrunert @hasheddan  @dims 
cc: @kubernetes/release-engineering

### CODE REVIEW

**Review by hasheddan — APPROVED** [review]
/lgtm

**Review by justaugustus — COMMENTED** [review]
/hold for thoughts

To satisfy some skew tests, I might suggest that we hold off removing a release's configs/jobs until the next release's EOL date.

i.e., don't remove 1.17's config until 2021-04-30 (1.18's EOL date)

cc: @kubernetes/sig-release-leads @kubernetes/release-engineering @kubernetes/sig-cluster-lifecycle @kubernetes/sig-cli-misc

**Comment by cpanato** [comment]
@justaugustus ok, I will just remove the milestone and merge warnings

**Comment by justaugustus** [comment]
> @justaugustus ok, I will just remove the milestone and merge warnings

SGTM. Can you also put this up as a topic for the next RelEng meeting?

**Comment by hasheddan** [comment]
> To satisfy some skew tests

Wouldn't the skew tests we would be concerned about be on the 1.18 boards? Is the idea to keep building 1.17 while 1.18 relies on it for skew?

**Comment by justaugustus** [comment]
> Is the idea to keep building 1.17 while 1.18 relies on it for skew?

Exactly! We've had issues in the past with this.
On my list this year (hopefully this cycle) is to solve for version markers, but in the meantime, we should try to avoid any breaking skew tests if we can.

**Comment by spiffxp** [comment]
I'm confused, why do we need to keep rebuilding 1.17 let alone run 1.17 jobs if we're not accepting any further changes to it?

Put another way, @justaugustus what issues have you had in the past related to skew testing that require this as a solution?

AFAICT the 1.17 jobs update the stable3 version marker, which isn't referenced in 1.18 jobs.

**Comment by spiffxp** [comment]
/hold
until above is addressed (am fine if another PR is opened for job changes or if job changes are added here)

**Comment by cpanato** [comment]
I will wait for a comment from @justaugustus I'm a bit confused as well

**Comment by justaugustus** [comment]
@cpanato -- Sorry for the back and forth. This is likely not a problem.
As this PR has a merge conflict, I've dropped the 1.17 jobs as part of https://github.com/kubernetes/test-infra/pull/20755.

Once this is rebased on top of that PR, we can get this approved.
/hold cancel

**Review by spiffxp — APPROVED** [review]
/approve
needs rebase but I trust you're not going to introduce any wild changes last-minute, ping if you need a re-lgtm

**Comment by cpanato** [comment]
@justaugustus @spiffxp PTAL thanks!

**Review by spiffxp — APPROVED** [review]
/approve
/lgtm
/hold cancel

**Pull request merged**
